src/imageio/format/tiff.c

```diff
@@ -874,7 +874,7 @@ void gui_init(dt_imageio_module_format_t *self)
 #endif
   const int compress = dt_conf_get_int("plugins/imageio/format/tiff/compress");
   const int compresslevel = dt_conf_get_int("plugins/imageio/format/tiff/compresslevel");
-  const int shortmode = dt_conf_get_int("plugins/imageio/format/tiff/shortfile");
+  const int shortmode = dt_conf_get_bool("plugins/imageio/format/tiff/shortfile");
```

Comment:
Not a regression on this PR so I'll merge but I found out that TIFF N&B doesn't work. The export is always in color with the exact same file size. Any idea?

Reply:
I'll check, maybe the monochrome test fails for some reason...

Reply:
Works for me as expected. However, I tested w/ e.g. real Pentax K3-III Monochrome DNG...

Reply:
Oh you mean this is only working if the image is already monochrome? Makes sense but was not clear to me.

Reply:
For TIFF, yes, there is no (defined) implicit conversion. And there is a check to first make sure the 3 channel are equal.

For AVIF it's different as it's converting internally to YUV w/ pre-defined coeffs, so we just keep luma.

That's why I'm insisting someone actually defines the requirements in https://github.com/darktable-org/darktable/issues/9266

Reply:
So, for grayscale TIFF export, you need to enable the monochrome module first explicitly if the image is a colour one...